components/bpp-recv/powerdown.c

```diff
@@ -6,14 +6,19 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <time.h>
 #include <sys/time.h>
+#include "esp_attr.h"
 #include "powerdown.h"
 
 typedef struct PowerItem PowerItem;
 static PowerDownCb *pdCb=NULL;
 static void *pdCbArg=NULL;
 static SemaphoreHandle_t mux;
 static TimerHandle_t tmr;
+static PowerMode powerMode;
+
+RTC_DATA_ATTR time_t savedWakeTimestamp[POWER_MODE_MAX];
```

Comment:
I prefer multiplies over divides and I prefer sub-second accuracy. There's enough rtc-memory.
We could use an unsigned long long and store everything in milliseconds.